cpp/tests/io/nested_json_test.cpp

```diff
@@ -395,110 +395,57 @@ TEST_F(JsonTest, UTF_JSON)
-    0x76, 0x65, 0x72, 0x73, 0x69, 0x6F, 0x6E, 0x22, 0x3A, 0x20, 0x22, 0x31, 0x2E, 0x34, 0x2E, 0x33,
-    0x22, 0x7D, 0x00, 0x29, 0x5C, 0x1C, 0x00, 0x00, 0x1C, 0x00, 0x00, 0x1C, 0x00, 0x00, 0x1C, 0x00,
-    0x00, 0x1C, 0x00, 0x00, 0x00, 0x0B, 0x04, 0x00, 0x00, 0x50, 0x41, 0x52, 0x31};
-
-  // Read in the data via parquet reader
-  cudf::io::parquet_reader_options read_opts = cudf::io::parquet_reader_options::builder(
-    cudf::io::source_info{reinterpret_cast<const char*>(parquet_data), sizeof(parquet_data)});
-  auto result = cudf::io::read_parquet(read_opts);
-
-  // Read in the data via the JSON parser
+  std::string const input = R"( [{"a":"0.0", "b":1.0}, {"b":1.1}, {"b":2.1, "a":"2.0"}] )";
+  // Get the JSON's tree representation
   auto const cudf_table = cuio_json::detail::parse_nested_json(
-    cudf::host_span<SymbolT const>{input.data(), input.size()}, default_options, stream);
+    cudf::host_span<SymbolT const>{input.data(), input.size()}, options, stream);
 
-  // Verify that the data read via parquet matches the data read via JSON
-  CUDF_TEST_EXPECT_TABLES_EQUAL(cudf_table.tbl->view(), result.tbl->view());
+  auto constexpr expected_col_count  = 2;
+  auto constexpr first_column_index  = 0;
+  auto constexpr second_column_index = 1;
+  EXPECT_EQ(cudf_table.tbl->num_columns(), expected_col_count);
 
-  // Verify that the schema read via parquet matches the schema read via JSON
-  cudf::test::expect_metadata_equal(cudf_table.metadata, result.metadata);
+  auto expected_col1 =
+    cudf::test::strings_column_wrapper({R"("0.0")", R"()", R"("2.0")"}, {true, false, true});
+  auto expected_col2            = cudf::test::fixed_width_column_wrapper<double>({1.0, 1.1, 2.1});
+  cudf::column_view parsed_col1 = cudf_table.tbl->get_column(first_column_index);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_col1, parsed_col1);
+  cudf::column_view parsed_col2 = cudf_table.tbl->get_column(second_column_index);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_col2, parsed_col2);
+}
+
+TEST_F(JsonTest, ExpectFailMixStructAndList)
+{
+  using cuio_json::SymbolT;
+
+  // Prepare cuda stream for data transfers & kernels
+  constexpr auto stream = cudf::default_stream_value;
+
+  // Default parsing options
+  cudf::io::json_reader_options options{};
+  options.enable_keep_quotes(true);
+
+  std::vector<std::string> const inputs{
+    R"( [{"a":[123], "b":1.0}, {"b":1.1}, {"b":2.1, "a":{"0":123}}] )",
+    R"( [{"a":{"0":"foo"}, "b":1.0}, {"b":1.1}, {"b":2.1, "a":[123]}] )",
+    R"( [{"a":{"0":null}, "b":1.0}, {"b":1.1}, {"b":2.1, "a":[123]}] )"};
+
+  for (auto const& input : inputs) {
+    // Get the JSON's tree representation
+    CUDF_EXPECT_THROW_MESSAGE(
+      auto const cudf_table = cuio_json::detail::parse_nested_json(
+        cudf::host_span<SymbolT const>{input.data(), input.size()}, options, stream),
+      "A mix of lists and structs within the same column is not supported");
```

Comment:
is it possible to print the input too with the error message?
```suggestion
      "A mix of lists and structs within the same column is not supported.") << " input:" << input;
```

Reply:
Unfortunately, `CUDF_EXPECT_THROW_MESSAGE` doesn't support piping further information.